### PR TITLE
Add notebook sample test: tfx sample

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/470)

### CODE REVIEW

**Comment by gaoning777** [comment]
solving https://github.com/kubeflow/pipelines/issues/263

File: test/sample-test/run_test.sh [hunk 1]
```diff
@@ -261,4 +265,52 @@ elif [ "$TEST_NAME" == "xgboost" ]; then
...
+  cd "${TEST_DIR}"
+  python3 check_notebook_results.py --experiment notebook-tfx-test --testname notebooktfx --result $SAMPLE_NOTEBOOK_TFX_TEST_RESULT --namespace ${NAMESPACE}
+
+  echo "Copy the test results to GCS ${RESULTS_GCS_DIR}/"
+  gsutil cp $SAMPLE_NOTEBOOK_TFX_TEST_RESULT ${RESULTS_GCS_DIR}/$SAMPLE_NOTEBOOK_TFX_TEST_RESULT
+
+  #Clean CMLE models
+  python3 clean_cmle_models.py --project ml-pipeline-test --model ${DEV_DEPLOYER_MODEL} --version ${MODEL_VERSION}
+  python3 clean_cmle_models.py --project ml-pipeline-test --model ${PROD_DEPLOYER_MODEL} --version ${MODEL_VERSION}
+elif [ "$TEST_NAME" == "notebook-lightweight" ]; then
```
Comment (Ark-kun) [inline test/sample-test/run_test.sh]: Thank's for testing the lightweight components notebook too!

Comment (gaoning777) [reply test/sample-test/run_test.sh]: BTW, this PR adds all the necessary codes, but the lightweight is not enabled yet. I'll send another PR to enable the lightweight notebook sample test.

File: samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb [hunk 2]
```diff
@@ -133,8 +145,13 @@
...
     "# Note that this notebook should be running in JupyterHub in the same cluster as the pipeline system.\n",
     "# Otherwise it will fail to talk to the pipeline system.\n",
+    "import kfp\n",
+    "from kfp import compiler\n",
+    "import kfp.dsl as dsl\n",
+    "import kfp.notebook\n",
+    "import kfp.gcp as gcp\n",
     "client = kfp.Client()\n",
-    "exp = client.create_experiment(name='demo')"
+    "exp = client.create_experiment(name=EXPERIMENT_NAME)"
```
Comment (Ark-kun) [inline samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb]: I think we should reuse the experiment if it already exists.

Comment (gaoning777) [reply samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb]: Yes. But for that, we can have another PR to do that. The scope for this PR is to add the sample tests, and it requires parameterizing the pipeline notebooks.

File: samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb [hunk 3]
```diff
@@ -22,6 +22,37 @@
...
+    "OUTPUT_DIR = 'Your-Gcs-Path' # Such as gs://bucket/objact/path\n",
+    "PROJECT_NAME = 'Your-Gcp-Project-Name'\n",
+    "BASE_IMAGE='gcr.io/%s/pusherbase:dev' % PROJECT_NAME\n",
+    "TARGET_IMAGE='gcr.io/%s/pusher:dev' % PROJECT_NAME\n",
+    "KFP_PACKAGE = 'https://storage.googleapis.com/ml-pipeline/release/0.1.3-rc.2/kfp.tar.gz'\n",
+    "TRAIN_DATA = 'gs://ml-pipeline-playground/tfx/taxi-cab-classification/train.csv'\n",
+    "EVAL_DATA = 'gs://ml-pipeline-playground/tfx/taxi-cab-classification/eval.csv'\n",
+    "HIDDEN_LAYER_SIZE = '1500'\n",
+    "STEPS = 3000\n",
+    "DATAFLOW_TFDV_IMAGE = 'gcr.io/ml-pipeline/ml-pipeline-dataflow-tfdv:0.1.3-rc.2'#TODO-release: update the release tag for the next release\n",
```
Comment (Ark-kun) [inline samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb]: Taking the images out of the components does not look great.
Can we do the following: The notebook compiles the pipeline as normal, but then the image buckets and tags are replaced. Similarly to how the sample tests are performed.

File: samples/notebooks/Lightweight Python components - basics.ipynb [hunk 4]
```diff
@@ -236,13 +250,7 @@
     "#Get or create an experiment and submit a pipeline run\n",
     "import kfp\n",
     "client = kfp.Client()\n",
-    "list_experiments_response = client.list_experiments()\n",
```
Comment (Ark-kun) [inline samples/notebooks/Lightweight Python components - basics.ipynb]: Why is this code being removed?

**Comment by IronPan** [comment]
I've looked into the pieces that touches the test workflow and permission. didnt looked into the test logic. 
/lgtm

File: samples/notebooks/Lightweight Python components - basics.ipynb (hunk 4 above)
Comment (gaoning777) [reply samples/notebooks/Lightweight Python components - basics.ipynb]: The test needs to know the experiment name in order to check the status for test results.

**Comment by Ark-kun** [comment]
/lgtm cancel

Waiting for review from @qimingj since the notebook was created by him.

**Comment by qimingj** [comment]
/lgtm

Awesome!

**Pull request merged**